benchmark/amazon_cellphones/simdjson_dom.h

```diff
@@ -0,0 +1,49 @@
+#pragma once
+
+#if SIMDJSON_EXCEPTIONS
+
+#include "amazon_cellphones.h"
+
+namespace amazon_cellphones {
+
+using namespace simdjson;
+
+struct simdjson_dom {
+  using StringType = std::string_view;
+
+  dom::parser parser{};
+
+  bool run(simdjson::padded_string &json, std::vector<brand<StringType>> &result) {
+    auto stream = parser.parse_many(json);
+    auto i = stream.begin();
+    ++i;  // Skip first line
+    for (;i != stream.end(); ++i) {
+      auto doc = *i;
+      auto x = result.begin();
+      while(true) {
+        if (x == result.end()) {
```

Comment:
Here `result` is a vector which accumulates all of the data parsed. If you only want an aggregated result, then it is wasteful.

(I will provide a code example soon.)